lib/widgets/content.dart

```diff
@@ -799,6 +801,90 @@ class MathBlock extends StatelessWidget {
   }
 }
 
+class MessageLinkPreview extends StatelessWidget {
+  const MessageLinkPreview({super.key, required this.node});
+
+  final LinkPreviewNode node;
+
+  @override
+  Widget build(BuildContext context) {
+    final messageListTheme = MessageListTheme.of(context);
+    final isSmallWidth = MediaQuery.sizeOf(context).width <= 576;
+
+    final dataContainer = Container(
```

Comment:
Is `dataContainer` the best name? I see the `Container` widget being used…what's the "data" and how does that widget "contain" it?

How about building this method's return value with help from a mutable `Widget result` variable? So here:

`Widget result = Container(/* etc. */);`

then below,

```dart
result = isSmallWidth
  ? Column(/* etc. */, children: [/* etc. */, result])
  : Row(/* etc. */, children: [/* etc. */, result]);
```

then `result = Container(decoration: /* etc. */, child: result);`

and finally `return result;`.